### PR TITLE
DC-1484: Enhance API

### PR DESCRIPTION
**Pull request by lxol**

### CODE REVIEW

File: app/uk/gov/hmrc/contactadvisors/controllers/SecureMessageController.scala [hunk 1]
```diff
@@ -71,24 +74,65 @@ class SecureMessageController @Inject()(customerAdviceAudit: CustomerAdviceAudit
     )
   }
 
+  def submitV2() =
+    Action.async { implicit request => {
+      Logger.info(s"******* V2RequestBody: ${request.body}")
```
Comment (andy1138) [inline app/uk/gov/hmrc/contactadvisors/controllers/SecureMessageController.scala]: If the body has 'personal' information then we cannot log it.

File: it/uk/gov/hmrc/contactadvisors/ApiISpec.scala [hunk 2]
```diff
@@ -0,0 +1,122 @@
+package uk.gov.hmrc.contactadvisors
```
Comment (andy1138) [inline it/uk/gov/hmrc/contactadvisors/ApiISpec.scala:17]: Copyright message missing

File: app/uk/gov/hmrc/contactadvisors/controllers/SecureMessageController.scala [hunk 3]
```diff
@@ -151,5 +213,30 @@ class CustomerAdviceAudit @Inject()(auditConnector: AuditConnector) {
...
+      res1.map {
+        case AdviceStored(messageId) => createEvent(Map("secureMessageId" -> messageId, "messageId" -> messageId), EventTypes.Succeeded, "Message Stored")
+        case _ => createEvent(Map("reason" -> s"Unexpected Error"), EventTypes.Failed, "Message Not Stored")
+      }.
+        recover { case ex =>
+          createEvent(Map("reason" -> s"Unexpected Error: ${ex.getMessage}"), EventTypes.Failed, "Message Not Stored")
+        }.
+        foreach { ev =>
+          auditConnector.sendEvent(ev).onFailure {
+            case err => Logger.error("Could not audit event", err)
```
Comment (andy1138) [inline app/uk/gov/hmrc/contactadvisors/controllers/SecureMessageController.scala]: - I dont know what in 'err' but should not have personal data.
- error messages will generate pagerduty events, is this what we want ?

Comment (lxol) [reply app/uk/gov/hmrc/contactadvisors/controllers/SecureMessageController.scala]: This is a replication of the existing API behaviour. The new ticket will address new API auditing.

**Pull request merged**